packages/core/src/interpreter.ts

```diff
@@ -842,17 +571,16 @@ export class Interpreter<
     };
   }
 
-  public [symbolObservable](): InteropSubscribable<
-    State<TContext, TEvent, TResolvedTypesMeta>
-  > {
+  public [symbolObservable](): InteropSubscribable<SnapshotFrom<TBehavior>> {
     return this;
   }
 
   public getSnapshot() {
```

Comment:
Currently, this doesn't have any semantic guarantees when it comes to the snapshot's identity. In practice, this expression won't always be true:
```ts
actorRef.getSnapshot() === actorRef.getSnapshot()
```

I think that it would be best to memorize this stuff at some level.

Reply:
We can shift the responsibility on the user though - it would be good to call out this in the docs though.